modules/gui.py

```diff
@@ -4444,6 +4445,63 @@ def setter_extra(_=None):
             imgui.end_table()
             imgui.spacing()
 
+        if draw_settings_section("Proxy"):
+            draw_settings_label(
+                "Type:",
+                "** Restart to apply changes **\n\nProxies with authentication aren't supported in login "
+                "dialog (SOCKS4 is not supported by Qt at all, SOCKS5 with authentication won't work and HTTP proxy "
+                "authentication just not implemented"
+            )
+            if imgui.begin_combo(f"###proxy_type", set.proxy_type.display):
+                for proxy_type in ProxyType:
+                    selected = proxy_type is set.proxy_type
+                    pos = imgui.get_cursor_pos()
+                    if imgui.selectable(f"###proxy_type_{proxy_type.name}", selected)[0]:
+                        set.proxy_type = proxy_type
+                        async_thread.run(db.update_settings("proxy_type"))
+                    if selected:
+                        imgui.set_item_default_focus()
+                    imgui.set_cursor_pos(pos)
+                    # self.draw_status_widget(proxy_type)
+                    imgui.same_line()
+                    imgui.text(proxy_type.display)
+                imgui.end_combo()
```

Comment:
if not using a custom widget for the combobox content (line draw_status_widget() does), might be able to just use a normal imgui combobox instead of the begin/end custom variant